README.md

```diff
@@ -20,3 +20,112 @@ This is just a template and needs some adaption to your specific use case.
+```
+
+Or you can build the container yourself from the [`./Dockerfile`](./Dockerfile):
+```bash
+# Execute in the repo's root dir:
+# (Please feel free to adapt the name/tag.)
+docker build -t ghga/my-microservice:<version> .
+```
+
+For production-ready deployment, we recommend using Kubernetes, however,
+for simple use cases, you could execute the service using docker
+on a single server:
+```bash
+# The entrypoint is preconfigured:
+docker run -p 8080:8080 ghga/my-microservice:<version>
+```
+
+If you prefer not to use containers, you may install the service from source:
+```bash
+# Execute in the repo's root dir:
+pip install .
+
+# to run the service:
+my-microservice
+```
+
+### Configuration:
+The [`./example-config.yaml`](./example-config.yaml) gives an overview of the available configuration options.
+Please adapt it, rename it to `.my_microservice.yaml`, and place it to one of the following locations:
+- in the current working directory were you are execute the service (on unix: `./.my_microservice.yaml`)
+- in your home directory (on unix: `~/.my_microservice.yaml`)
+
+The config yaml will be automatically parsed by the service.
+
+**Important: If you are using containers, the locations refer to paths within the container.**
+
+All parameters mentioned in the [`./example-config.yaml`](./example-config.yaml)
+could also be set using environment variables or file secrets.
+
+For naming the environment variables, just prefix the parameter name with `MY_MICROSERVICE_`,
+e.g. for the `host` set an environment variable named `MY_MICROSERVICE_HOST`
+(you may use both upper or lower cases, however, it is standard to define all env
+variables in upper cases).
+
+To using file secrets please refer to the
+[corresponding section](https://pydantic-docs.helpmanual.io/usage/settings/#secret-support)
+of the pydantic documentation.
+
+
+## Development
+For setting up the development environment, we rely on the
+[devcontainer feature](https://code.visualstudio.com/docs/remote/containers) of vscode
+in combination with Docker Compose.
+
+To use it, you have to have Docker Compose as well as vscode with its "Remote - Containers" extension (`ms-vscode-remote.remote-containers`) extension installed.
+Then, you just have to open this repo in vscode and run the command
+`Remote-Containers: Reopen in Container` from the vscode "Command Palette".
+
+This will give you a full-fledged, pre-configured development environment including:
+- infrastructural dependencies of the service (databases, etc.)
+- all relevant vscode extensions pre-installed
+- pre-configured linting and auto-formating
+- a pre-configured debugger
+- automatic license-header insertion
+
+Moreover, inside the devcontainer, there are two convenience commands available
+(please type them in the integrated terminal of vscode):
+- `dev_install` - install the service with all development dependencies,
+installs pre-commit, and applies any migration scripts to the test database
```

Comment:
BTW, why is pre-commit not installed in the Dockerfile?

Reply:
The workspace is mounted after the container builds. And precommit installs as a git hook to the .git directory.

Reply:
👍